drivers/storage/portworx/components_test.go

```diff
@@ -265,6 +266,71 @@ func TestBasicComponentsInstall(t *testing.T) {
 	require.Equal(t, 2, len(statefulset.Spec.Template.Spec.Containers))
 }
 
+func TestPxRepoInstallUninstall(t *testing.T) {
+	coreops.SetInstance(coreops.New(fakek8sclient.NewSimpleClientset()))
+	reregisterComponents()
+	k8sClient := testutil.FakeK8sClient()
+	driver := portworx{}
+	driver.Init(k8sClient, runtime.NewScheme(), record.NewFakeRecorder(0))
+
+	cluster := &corev1.StorageCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "px-cluster",
+			Namespace: "kube-test",
+		},
+		Spec: corev1.StorageClusterSpec{
+			PxRepo: &corev1.PxRepoSpec{
+				Enabled: false,
+				Image:   "testImage",
+			},
+		},
+	}
+
+	// Disabled first
+	cluster.Spec.PxRepo.Enabled = false
+	err := driver.PreInstall(cluster)
+	require.NoError(t, err)
+
+	service := &v1.Service{}
+	err = testutil.Get(driver.k8sClient, service, component.PxRepoServiceName, cluster.Namespace)
+	require.True(t, errors.IsNotFound(err))
+
+	deployment := &appsv1.Deployment{}
+	err = testutil.Get(driver.k8sClient, deployment, component.PxRepoDeploymentName, cluster.Namespace)
+	require.True(t, errors.IsNotFound(err))
+
+	// Enable the service
+	cluster.Spec.PxRepo.Enabled = true
+	err = driver.PreInstall(cluster)
+	require.NoError(t, err)
+
+	err = testutil.Get(driver.k8sClient, service, component.PxRepoServiceName, cluster.Namespace)
+	require.NoError(t, err)
+
+	err = testutil.Get(driver.k8sClient, deployment, component.PxRepoDeploymentName, cluster.Namespace)
+	require.NoError(t, err)
+
+	// Change the deployment
+	cluster.Spec.ImagePullPolicy = v1.PullNever
+	err = driver.PreInstall(cluster)
+	require.NoError(t, err)
+
+	err = testutil.Get(driver.k8sClient, deployment, component.PxRepoDeploymentName, cluster.Namespace)
+	require.NoError(t, err)
+	require.Equal(t, deployment.Spec.Template.Spec.Containers[0].ImagePullPolicy, v1.PullNever)
+
+	// Disable the service
+	cluster.Spec.PxRepo.Enabled = false
+	err = driver.PreInstall(cluster)
+	require.NoError(t, err)
+
+	err = testutil.Get(driver.k8sClient, service, component.PxRepoServiceName, cluster.Namespace)
+	require.True(t, errors.IsNotFound(err))
+
+	err = testutil.Get(driver.k8sClient, deployment, component.PxRepoDeploymentName, cluster.Namespace)
+	require.True(t, errors.IsNotFound(err))
+}
+
```

Comment:
Can we add px-repo to the following tests for testing change in image, pull secret, affinity or tolerations -
https://github.com/libopenstorage/operator/blob/master/drivers/storage/portworx/components_test.go#L5871
https://github.com/libopenstorage/operator/blob/master/drivers/storage/portworx/components_test.go#L6596
https://github.com/libopenstorage/operator/blob/master/drivers/storage/portworx/components_test.go#L7454
https://github.com/libopenstorage/operator/blob/master/drivers/storage/portworx/components_test.go#L7759
https://github.com/libopenstorage/operator/blob/master/drivers/storage/portworx/components_test.go#L8150